src/fu-remotes.c

```diff
@@ -0,0 +1,503 @@
+/*
+ * Copyright (C) 2017-2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN				"FuRemotes"
+
+#include "config.h"
+
+#include <xmlb.h>
+#include <glib-object.h>
+#include <gio/gio.h>
+#include <glib/gi18n.h>
+
+#include "fu-common.h"
+#include "fu-remotes.h"
+
+#include "fwupd-common.h"
+#include "fwupd-error.h"
+#include "fwupd-remote-private.h"
+
+enum {
+	SIGNAL_CHANGED,
+	SIGNAL_LAST
+};
+
+static guint signals[SIGNAL_LAST] = { 0 };
+
+static void fu_remotes_finalize	 (GObject *obj);
+
+struct _FuRemotes
+{
+	GObject			 parent_instance;
+	GPtrArray		*remotes;		/* (element-type FwupdRemote) */
+	GPtrArray		*monitors;		/* (element-type GFileMonitor) */
+	XbSilo			*silo;
+	GHashTable		*os_release;
+};
+
+G_DEFINE_TYPE (FuRemotes, fu_remotes, G_TYPE_OBJECT)
+
+static void
+fu_remotes_emit_changed (FuRemotes *self)
+{
+	g_debug ("::remotesuration changed");
+	g_signal_emit (self, signals[SIGNAL_CHANGED], 0);
+}
+
+static GPtrArray *
+fu_remotes_get_all_paths (void)
+{
+	GPtrArray *paths = g_ptr_array_new_with_free_func (g_free);
+	const gchar *remotes_dir;
+	const gchar *system_prefixlibdir = "/usr/lib/fwupd";
```

Comment:
I know this was hardcoded before, but I think that it should actually be added in via meson define.

Reply:
This is actually deliberate. We add the ones in FU_PATH_KIND_SYSCONFDIR_PKG first, and then fall back to the per-system ones. This means if you run fwupd in a development prefix you still pick up the system ones so that the daemon actually works properly.

Reply:
Even with systemd these days? Systemd environment overrides work without this I think.

Reply:
Lets keep the behaviour the same for this PR and then try to rip the manual path out in a follow up PR.